### PR TITLE
Add stream scheduler `transfer_when_all` tests

### PR DESCRIPTION
**Pull request by trxcllnt**

### CODE REVIEW

File: test/nvexec/transfer_when_all.cpp [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  (void)snd;
+}
+
+// Hangs in `sync_wait`
+// TEST_CASE("transfer_when_all with no senders", "[cuda][stream][adaptors][transfer_when_all]") {
+//   nvexec::stream_context stream_ctx{};
+//   auto gpu = stream_ctx.get_scheduler();
+//   auto snd = ex::transfer_when_all(gpu);
+//   wait_for_value(std::move(snd));
+// }
```
Comment (ericniebler) [inline test/nvexec/transfer_when_all.cpp:50]: @trxcllnt is this bug filed. @senior-zero are you aware of this?

Comment (trxcllnt) [reply test/nvexec/transfer_when_all.cpp:50]: @ericniebler Not yet, I just discovered it last night

Comment (trxcllnt) [reply test/nvexec/transfer_when_all.cpp]: This is fixed.

**Review by senior-zero — APPROVED** [review]

**Comment by ericniebler** [comment]
Thank you @trxcllnt !

**Pull request merged**